### ISSUE TITLE
Add to Multiple New Hits

### ISSUE THREAD

**Issue by Bella-Roma** [oswald2/DGPatchMaker#11]
System: Linux Mint 19.2 MATE 
DGPatchmaker: Most current git master (tried on the Ubuntu 18.04 binary version as well)

Hi! 

I'm- not sure if the following is an issue or if I'm doing something wrong. 
I have created a Drum Gizmo kit using DGPatchMaker before, so I am familiar with the process, however, I have never worked with a drum kit/library with this many sample files before. I'll try and be as descriptive as I can in regards to my work flow.

I just purchased a large Drum Drops kit (60's Motown Kit). The kit has multiple drum types, drum hits, and 3 Round Robin stereo files per velocity, so that in total there is roughly 2800 sample files for the entire Motown library. I'm making this into a Drum Gizmo kit manually with DGPatchMaker because all of the current Drum Drop parsers report back errors and end up importing empty instrument files. It looks like this is due to the naming conventions DD chose for this particular kit.

When adding the instruments, I fill in the instrument's info fields, then add a 'hit' in the left-hand pane. I then import the samples for that particular drum in the right-hand pane. After that I choose the channel routing (Channel 1 = Snare L and Channel 2 = Snare R for example). 

Next I start selecting both the Left and Right channels of a particular Round Robin velocity sample file and choose 'Add as new hit' from the context menu. As there are a total of 3 Round Robin stereo files per each velocity, the left-hand pane in DGPatchMaker shows the following hierarchy for just one velocity range (power hit rating 4.0):

> **LHWS-Snare-Center-1            4.0**
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR01 Snare L
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR01 Snare R
> **LHWS-Snare-Center-2            4.0**
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR02 Snare L
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR02 Snare R
> **LHWS-Snare-Center-3            4.0**
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR03 Snare L
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR03 Snare R

Adding 2 channels at a time will take a very long time (there are 300 snare wav files alone!). I thought that if I selected all three Round Robin stereo files for a particular velocity and then chose "Add to Multiple New hits", that DGPatchMaker would then create individual 'hits' for each pair of L&R channels automatically. But what I see happening is that it's creating a new hit for every 3 pairs of Round Robin files of a particular velocity range, so that the left pane is structured something like this:

> **LHWS-Snare-Center-1          4.0**
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR01 Snare L
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR01 Snare R
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR02 Snare L
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR02 Snare R
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR03 Snare L
> 	DD020_LWHS_SNARE_CENTRE_MIX_V01_RR03 Snare R

I've found it isn't always consistent either. Sometimes it adds the 3 related stereo files to a single new hit, sometimes it doesn't. It's quite possible I am misunderstanding the 'add to multiple new hits' function though. From what I think I understand, each Round Robin variant of a particular velocity range should be a separate 'hit' but must share the same power rating - not combined into one hit like the above - is that correct?

My ultimate goal is to try to further simplify the process if possible, given the sheer number of samples in this Drum Drops bank. I previously used the import feature in DGPatchMaker on the Drum Drops 1965 Ludwig Super Classic Kit and it went very well, but this Motown kit was problematic. I wish their was a more consistent naming convention across drum kit products.

Anyway, DGP is great and it helps a lot :)

**Comment by oswald2** [comment]
Yes, I think there is a bug still lurking there, or more a quirky implementation that doesn't do what you expect. I need to have a more detailed look.

I am currently on a Hackathon this weekend, so it might take some time that I have a look at it, but I will. Also some new features will be out soon, supporting the new features of DrumGizmo.

**Comment by Bella-Roma** [comment]
Hi - just wanted to update this topic as I figured out a way to correctly add samples to multiple new hits, and it's been pretty consistent:

If I select a pair of channels for a sample file, like so:
![Selection_002](https://user-images.githubusercontent.com/60560450/84561453-5d4b2600-ad01-11ea-8be3-88fe1a5c9ad9.png)

then right click and choose "Add to multiple new hits", the desired behaviour does occurs and all other channel pairs in the 'Sample' pane are added to their own individual hit in the 'Hits' pane:
![Selection_004](https://user-images.githubusercontent.com/60560450/84561478-98e5f000-ad01-11ea-8404-bbe0f98e11c9.png)

This saved me a ton of time and work!

I do still have to go in and manually input matching velocity/hit power values for Round Robin groups because there is an issue with the calculated hit power feature, but I'll add that to it's own ticket shortly.

Thanks!
